test/object-store/sync/flx_role_change.cpp

```diff
@@ -802,57 +813,403 @@ TEST_CASE("flx: role changes during bootstrap complete successfully", "[sync][fl
+                if (error.status != ErrorCodes::SyncClientResetRequired)
+                    FAIL(util::format("Unexpected error received by error handler: %1", error.status));
+            };
+        };
+
+        // Start with the role/rules set to allow only manager & director records
+        update_role(default_rule, {{"role", {{"$in", {"manager", "director"}}}}});
+        logger->debug("ROLE CHANGE: Initial rule definitions: %1", default_rule);
+        REQUIRE(app_session.admin_api.update_default_rule(app_session.server_app_id, default_rule));
+
+        auto config_1 = harness->make_test_file();
+        auto&& [reset_future, reset_handler] = reset_utils::make_client_reset_handler();
+        config_1.sync_config->notify_after_client_reset = std::move(reset_handler);
+        config_1.sync_config->client_resync_mode = ClientResyncMode::Recover;
+        setup_config_callbacks(config_1);
+
+        auto realm_1 = Realm::get_shared_realm(config_1);
+        {
+            // Set up a default subscription for all records of the Person class
+            auto table = realm_1->read_group().get_table("class_Person");
+            auto new_subs = realm_1->get_latest_subscription_set().make_mutable_copy();
+            new_subs.clear();
+            new_subs.insert_or_assign(Query(table));
+            auto subs = new_subs.commit();
+            subs.get_state_change_notification(sync::SubscriptionSet::State::Complete).get();
+            wait_and_verify(realm_1, 0, params.num_mgrs, params.num_dirs);
+        }
+        // The test will update the rule to change access from only manager and director records
+        // to only the employee records while a client reset is in progress.
+        update_role(default_rule, {{"role", "employee"}});
+        // Force a client reset to occur the next time the session connects
+        reset_utils::trigger_client_reset(app_session, realm_1);
+
+        // Each one of these sections runs the role change client reset test with the different
+        // setting for the `update_role_state` which indicates which stage during the client reset
+        // where the role change will occur.
+        SECTION("Bind before client reset") {
+            logger->debug("ROLE CHANGE: Role change after BIND before client reset");
+            // Trigger the role change just after the BIND message is sent prior to receiving
+            // the client reset error - don't check for the role change (200) error, since it
+            // is unreliable to catch before the restart to perform the client reset.
+            setup_test_params(ClientResetTestState::bind_before_cr_session, true);
+        }
+        SECTION("Client reset session ident") {
+            logger->debug("ROLE CHANGE: Role change after client reset session IDENT");
+            // Trigger the role change just after the IDENT message is sent for the fresh realm
+            // download session.
```

Comment:
This is three different descriptions of what the test is testing, with the most important one being the least useful description. It's okay for SECTION titles to be pretty long-winded, and if you're finding it useful to log information in each section it's a pretty strong sign that you should just put that information in the section name. Some of these comments have some details that make sense as a comment, but for a lot of them a slightly reworded version of the comment should just be the section name.

Reply:
I cleaned up the section names and the comments - I left the (redundant) debug statements in since it makes it easier to track down which test was being run when the failure occurred - but they have been updated to return the same text as the section name.